### ISSUE TITLE
Extend the tabular pipeline and run a model-family ablation matrix

### ISSUE THREAD

**Issue by annawoodard** [dsi-clinic/vanguard#116]
<html><head></head><body><h2>Background</h2>
<p>The tabular feature pipeline already exists. This subproject is <strong>not</strong> about inventing new features — it's about comparing <strong>model families</strong> fairly while keeping the feature set fixed. Otherwise you can't tell whether a gain came from features or from the model.</p>
<p>The repo already has <strong>feature-arm ablations</strong> (which feature blocks carry signal?). This issue adds <strong>model-family ablations</strong> (which model family makes the best use of those blocks?).</p>
<h3>What already exists</h3>

File | Role
-- | --
train_tabular.py | Main tabular training entrypoint
tabular_models.py | Current sklearn model builder (logistic regression, random forest)
evaluation/ | Shared fold generation, metrics, saved outputs
configs/ispy2.yaml | Standard tabular run config
configs/independent_signal.yaml | Focused comparison against the clinical + tumor_size baseline

<p>The pipeline already supports numeric feature selection, nested tuning for logistic regression, stratified CV, site-exclusive group-aware CV, and per-stratum metrics.</p>
<h2>Key Definitions</h2>
<p><strong>Model selection</strong> — Try a small number of reasonable setups, compare them fairly, and choose based on training-data rules — not by hand-picking the best validation result after the fact.</p>
<p><strong>Nested tuning</strong> — Outer folds estimate real performance; inner folds choose hyperparameters. Safer than tuning and reporting on the same fold.</p>
<p><strong>Robustness</strong> — The result still holds under reasonable changes: site-exclusive folds, unilateral-only cohort, performance broken out by tumor subtype.</p>
<p><strong>Baseline arm</strong> — A simpler reference model (here, <code>clinical + tumor_size</code>) so you can ask "how much better is this than something simple?"</p>
<p><strong>Model-family ablation</strong> — Fix the feature arm, swap the model family, compare. Example: run the same <code>clinical + tumor_size + graph + kinematic</code> arm with logistic regression, random forest, and XGBoost.</p>
<h2>What To Do</h2>
<h3>1. Orient yourself</h3>
<p>Read the current tabular code and write a short summary of:</p>
<ul>
<li>which model families already exist</li>
<li>which feature-arm configs already exist</li>
<li>what result summaries are already tracked</li>
</ul>
<h3>2. Add at least one new model family</h3>
<p><code>xgboost</code> is already in <code>requirements.txt</code>, so adding a boosting model is a natural next step.</p>
<p><strong>Recommended set:</strong> logistic regression, random forest, XGBoost.
<strong>Optional extras if time allows:</strong> histogram gradient boosting, extra trees.</p>
<h3>3. Run a focused model-family × feature-arm matrix</h3>
<p>Start small — don't use every arm. A good starting set:</p>
<ul>
<li><code>clinical + tumor_size</code> (baseline)</li>
<li>Best single vessel arm (e.g. <code>clinical + tumor_size + kinematic</code>)</li>
<li>Strongest combined vessel arm (e.g. <code>clinical + tumor_size + morph + graph + kinematic</code>)</li>
</ul>
<p>Expand later if compute allows.</p>
<h3>4. Keep evaluation consistent and save a merged result table</h3>
<p>One table with columns: arm name, model family, feature-selection mode, nested tuning on/off, AUC mean/std, AP mean/std.</p>
<p><strong>Submit non-trivial CV runs through Slurm, not the headnode.</strong></p>
<h2>Guidance</h2>
<p>Try to answer:</p>
<ul>
<li>Does a stronger nonlinear model beat logistic regression?</li>
<li>On all arms, or only on richer vessel arms?</li>
<li>Is the extra complexity worth it?</li>
</ul>
<p><strong>Don't mix too many variables at once.</strong> First compare model families. Then compare tuning choices within the top one or two families.</p>
<p>If two setups are very close, prefer the simpler, easier-to-explain, more stable one.</p>
<h2>Definition of Done</h2>
<ul>
<li>Pipeline supports at least one additional model family beyond the current two</li>
<li>A model-family × feature-arm comparison table exists</li>
<li>Runs are reproducible from saved YAML configs</li>
<li>One or two best families are chosen for the rest of Project 2, with a short note explaining why</li>
</ul></body></html>
